### PR TITLE
Developer Tools page

### LINKED ISSUE #2035: Create Developer Tools landing page
This page should:
* Serve as a landing page for two others: the API status page and the Simulations debug page
* Mirror the design language of all of our other static pages. This includes a consistent page gutter, standard margins, and the use of styling constants defined in `src/styles`

It should not be integrated into the navbar. Instead, it should be linked into the standard page footer. Separately, as part of #2032, it should be added to the policy calculator footer that should now be enabled while the policy simulation loads.

### PR DESCRIPTION
**Pull request by bharateshwq**
## Description

Fixes #2035

Created basic wire frame design for how the developer tools should look like while keeping in consideration the design language of policy Engine
Let me know if there are changes required for this

This in no way is the final commit that i expect to be merged there are a lot of optimization, responsiveness tasks and restructuring to be 

## Screenshots

image

### CODE REVIEW

**Comment by bharateshwq** [comment]
Hey @MaxGhenis , 
I have pushed a new commit can you have a look at it

**Comment by anth-volk** [comment]
@bharateshwq This is looking great! Thanks for your work on it. Could you just do the following on the design:

* Pad the titles underneath the images more on the top, perhaps to about 1rem
* Show how this looks on tablet (769 to ~1200px) and mobile (<769px) breakpoints

I might also ask for a couple changes to the text of the header afterward

**Comment by bharateshwq** [comment]
Hey @anth-volk,
I have done the necessary changes and added responsiveness to the developer-hub page

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks for your work on this @bharateshwq, this is looking good. There are a few changes I was hoping to request:

* At the moment, this code places a wrapper around the API status and simulations pages. Could you remove that, including the "back" arrow? The pages themselves should actually be unchanged.
* Could you link this page into the footer with the tag "Developer tools"?
* Minor, but could you add padding to the right of the text within the link boxes at the mobile viewpoint? In the image below, you can see how the word "Simulations" runs into the button.
Screen Shot 2024-10-02 at 5 57 36 PM

I might also request some textual changes, but am hoping to do that at the end.

**Comment by bharateshwq** [comment]
Hey @anth-volk, 

I'll definitely fix that. Do you prefer having the `api-status` route separate, or keeping it under `developer-tools/api-status`? Right now, it's the latter, in which case users might find it more intuitive to have a back button on the API status and simulation pages. Let me know your thoughts!

**Comment by anth-volk** [comment]
@bharateshwq I think the way you layered the URLs makes sense, but I also think their browser back button should be able to handle that navigation action, as I believe `react-router-dom` writes to history.

**Comment by bharateshwq** [comment]
Yes @anth-volk, React Router does write to the history, so using the browser's back button would work for navigation. However, I'll be removing the back button from the UI. Additionally, the current(master) API status and simulation headers  have different designs, which I feel may not appear consistent to the user.
How would you like me to proceed with this?

**Comment by anth-volk** [comment]
@bharateshwq That is a good point. Considering these are internal developer tools, I don't think it's too big of an issue, but certainly warrants opening a separate issue. I can't remember their designs well enough, but I'm guessing the API status page is more out-of-line with our current design. If you'd like to take it on after this one, we'd greatly appreciate it.

**Comment by bharateshwq** [comment]
Hey @anth-volk, I've made the updates as requested and am awaiting your feedback. Thanks!

**Comment by anth-volk** [comment]
@bharateshwq Thanks, I'm hoping to get to this either tonight or tomorrow

File: src/layout/Footer.jsx
Comment (anth-volk) [inline src/layout/Footer.jsx:66]:
```suggestion
      label: "Developer Tools",
```

File: src/pages/DeveloperHome.jsx
Comment (anth-volk) [inline src/pages/DeveloperHome.jsx:14]:
```suggestion
```

File: src/pages/DeveloperHome.jsx
Comment (anth-volk) [inline src/pages/DeveloperHome.jsx:100]:
```suggestion
```

File: src/pages/DeveloperHome.jsx
Comment (anth-volk) [inline src/pages/DeveloperHome.jsx:138]: Just personal preference: structs do exist in JS (albeit more as a concept) could you change this name?

File: src/pages/DeveloperHome.jsx
Comment (anth-volk) [inline src/pages/DeveloperHome.jsx:186]:
```suggestion
```

File: src/pages/DeveloperLayout.jsx
Comment (anth-volk) [inline src/pages/DeveloperLayout.jsx:32]:
```suggestion
            Welcome to the Developer Tools page for PolicyEngine. This hub is
```
Having fun is not allowed in tax and benefit simulation

**Review by anth-volk — CHANGES_REQUESTED** [review]
Hey @bharateshwq. Thanks again for your work on this, this will be a major benefit for internal development. I've left a couple of minor change requests, and also took the liberty of cleaning up a merge conflict/linting. After this, we should be good to merge!

**Comment by bharateshwq** [comment]
Hey @anth-volk ,
sorry for the late reply i have sent a new commit with all suggested changes

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks for your work on this @bharateshwq, it's looking great.

Earlier, when I had asked that you remove the enclosing features, it was because it created a wrapper that added a "back" button and created two footers. We actually still want the "header" (the navbar) and the footer, just not the back button. Could you add the navbar back into these two pages?

After that, this will be good to go.

**Comment by bharateshwq** [comment]
Hey,
i have made a new commit fixing this

**Comment by anth-volk** [comment]
@bharateshwq I've taken the liberty of addressing some merge conflicts in the PR. I'll approve and merge once tests pass.

**Review by anth-volk — APPROVED** [review]
Thank you very much @bharateshwq! Merging now.

**Pull request merged**
